project_a/CMakeLists.txt

```diff
@@ -1,22 +1,26 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.11)
 project(ProjectA VERSION 0.1)
 enable_language(CXX)
 
-# build the library
-add_library(project_a
+# to make things complicated, first build an object library
+add_library(project_a_objects OBJECT 
         src/private/project_a_private.cpp
         src/public/project_a_public.cpp
         )
 
-target_include_directories(project_a
+target_include_directories(project_a_objects
         PUBLIC
             $<INSTALL_INTERFACE:include>
```

Comment:
For IDEs I want to have the possibility to point to the `${CMAKE_CURRENT_SOURCE_DIR}/include`, i.e. in the build folder.